packages/mui-material/src/Tabs/Tabs.js

```diff
@@ -375,9 +375,10 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
     scrollbarWidth: 0,
   });
 
-  const valueToIndex = new Map();
+  const valueToIndex = React.useMemo(() => new Map(), []);
```

Comment:
Should we use a ref here instead?

Reply:
Makes sense. While useMemo technically works here, it is meant for **derived values** based on dependencies. Updated in https://github.com/mui/material-ui/pull/46333/commits/fd86ebd2caee0137c24e72eb2d5c83706759a317.